compiler/rustc_middle/src/lint.rs

```diff
@@ -90,8 +90,20 @@ pub fn reveal_actual_level(
     // future compatibility warning.
     if level == Level::Warn && lint != LintId::of(FORBIDDEN_LINT_GROUPS) {
         let (warnings_level, warnings_src) = probe_for_lint_level(LintId::of(builtin::WARNINGS));
-        if let Some(configured_warning_level) = warnings_level {
-            if configured_warning_level != Level::Warn {
+        if let Some(configured_warning_level) = warnings_level
+            && configured_warning_level != Level::Warn
+        {
+            if matches!(warnings_src, LintLevelSource::CommandLine(..))
+                && lint != LintId::of(builtin::WARNINGS)
+                && let (Some(lint_level), lint_src) = probe_for_lint_level(lint)
```

Comment:
Those are `level` and `src` arguments to the current function. We shouldn't recompute them.